applications/asset_tracker_v2/src/modules/Kconfig.location_module

```diff
@@ -30,14 +30,8 @@ config LOCATION_MODULE_NEIGHBOR_CELLS_DATA_CONVERT_RSRQ_TO_DB
 	  using the nRF Cloud cellular positioning library.
 
 # The cloud interface for A-GPS, P-GPS and cell location is handled by the application
-config LOCATION_METHOD_GNSS_AGPS_EXTERNAL
-	default y if NRF_CLOUD_AGPS
-
-config LOCATION_METHOD_GNSS_PGPS_EXTERNAL
-	default y if NRF_CLOUD_PGPS
-
-config LOCATION_METHOD_CELLULAR_EXTERNAL
-	default y
+config LOCATION_SERVICE_EXTERNAL
+	default y if NRF_CLOUD_AGPS || NRF_CLOUD_PGPS || LOCATION_METHOD_CELLULAR || LOCATION_METHOD_WIFI
```

Comment:
Is this necessary, or would just a select under `LOCATION_MODULE` be enough? Or is it needed for GNSS only configuration without assistance?

Reply:
Yep, plain GNSS would require this although I think ATv2 is generally not build time configurable for disabling neither cellular or GNSS.

@simensrostad Can you comment whether I just make this LOCATION_EXTERNAL selection without conditions, i.e., do we want to save flash if built without cellular positioning?

Reply:
I tried compiling with all above options disabled, but it doesn't even seem to work:

`west build -b nrf9160dk_nrf9160_ns -p -- -DCONFIG_NRF_CLOUD_AGPS=n -DCONFIG_NRF_CLOUD_PGPS=n -DCONFIG_LOCATION_METHOD_CELLULAR=n -DCONFIG_LOCATION_METHOD_WIFI=n`

```
/home/tok1/work/ncs/nrf/applications/asset_tracker_v2/src/modules/location_module.c: In function 'location_event_handler':
/home/tok1/work/ncs/nrf/applications/asset_tracker_v2/src/modules/location_module.c:494:64: error: 'const struct location_event_data' has no member named 'cellular_request'
  494 |                         (struct lte_lc_cells_info *)&event_data->cellular_request);
      |
```
Is it necessary to even support this? IMHO, you could just select `LOCATION_SERVICE_EXTERNAL` whenever the Location module is enabled.

Reply:
I'm not sure if A-GPS can even be disabled. Since we've treated it as a default feature. Since A-GPS is always on anyways I guess `LOCATION_SERVICE_EXTERNAL` can always be y selected.

Reply:
> Looking at the code, cellular isn't flagged. Should it be possible to disable cellular positioning?

ATv2 hasn't had support for not compiling GNSS or cellular positioning functionality into the flash. There are both build and run time configurations for enabling/disabling sampling of GNSS and cellular data but it's been always in the image. Hence, I haven't made efforts to enable compilation of ATv2 without cellular positioning although I tried and that wouldn't require much flagging.

> I'm not sure if A-GPS can even be disabled. Since we've always treated it as a default feature. If A-GPS is always on anyways I guess `LOCATION_SERVICE_EXTERNAL` can always be y selected.

I remember @coderbyheart mentioning that it's a valid configuration to disable A-GPS and I made configuration for it into `sample.yaml` ensuring it doesn't get broken easily. 

I will remove redefinition of `LOCATION_SERVICE_EXTERNAL` and select this in `LOCATION_MODULE` definition.